src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -488,6 +509,14 @@ private static void ThrowIfCURLMError(int error)
             }
         }
 
+        private static bool AreEqualNetworkCredentials(NetworkCredential credential1, NetworkCredential credential2)
+        {
+            Debug.Assert(credential1 != null && credential2 != null, "arguments are non-null in network equality check");
+            return credential1.UserName == credential2.UserName &&
+                credential1.Domain == credential2.Domain &&
+                string.Equals(credential1.Password, credential2.Password, StringComparison.Ordinal);
```

Comment:
The equality operator on string uses an ordinal comparison.  Why is the password comparison done differently from the username comparison and the domain comparison?